account_cutoff_accrual_picking/models/account_cutoff.py

```diff
@@ -0,0 +1,281 @@
+                currency = order.currency_id
+                analytic_account_id = order.project_id.id
+                taxes = order_line.tax_id
+                account = product.product_tmpl_id.\
+                    _get_product_accounts()['income']
+                if not account:
+                    raise UserError(_(
+                        "Missing income account on product '%s' or on its "
+                        "related product category '%s'.") % (
+                            product.display_name,
+                            product.categ_id.display_name))
+                account_id = order.fiscal_position_id.map_account(account).id
+
+        oline_dict[order_line].update({
+            'price_unit': price_unit,
+            'price_origin': price_origin,
+            'currency': currency,
+            'analytic_account_id': analytic_account_id,
+            'account_id': account_id,
+            'taxes': taxes,
+            })
+
+    def stock_move_update_oline_dict(self, move_line, oline_dict):
+        dpo = self.env['decimal.precision']
+        qty_prec = dpo.precision_get('Product Unit of Measure')
+        if self.type == 'accrued_expense':
+            if (
+                    move_line.purchase_line_id and
+                    move_line.purchase_line_id not in oline_dict and
+                    not float_is_zero(
+                        move_line.purchase_line_id.product_qty,
+                        precision_digits=qty_prec)):
+                self.order_line_update_oline_dict(
+                    move_line.purchase_line_id, 'purchase', oline_dict)
+        elif self.type == 'accrued_revenue':
+            if (
+                    move_line.procurement_id and
+                    move_line.procurement_id.sale_line_id and
+                    move_line.procurement_id.sale_line_id not in oline_dict and
+                    not float_is_zero(
+                        move_line.procurement_id.sale_line_id.product_uom_qty,
+                        precision_digits=qty_prec)):
+                self.order_line_update_oline_dict(
+                    move_line.procurement_id.sale_line_id, 'sale', oline_dict)
+
+    def get_lines(self):
+        res = super(AccountCutoff, self).get_lines()
+        spo = self.env['stock.picking']
+        aclo = self.env['account.cutoff.line']
+        acmo = self.env['account.cutoff.mapping']
+
+        pick_type_map = {
+            'accrued_revenue': 'outgoing',
+            'accrued_expense': 'incoming',
+            }
+        cutoff_type = self.type
+        if cutoff_type not in pick_type_map:
+            return res
+
+        # Create account mapping dict
+        account_mapping = acmo._get_mapping_dict(
+            self.company_id.id, cutoff_type)
+
+        # TODO date_done is a Datetime field, so maybe we need more clever code
+        # for our friends which are far away from GMT
+        pickings = spo.search([
```

Comment:
Technically you could have a move done without having the picking completely done. I suggest to analyze move instead of picking.
Functionally, I'm really concerned by the fact that the moves related to expenses/revenues can be of both type incoming and outgoing. For instance, for SO, you have outgoing moves for deliveries but incoming moves for returns.